File: pa-04/frontend/src/backend/bidService.js
import {config, appFetch} from './appFetch';

export const bidOnAProduct = (productId, maxBid,
                              onSuccess, onErrors) =>
    appFetch(`/bid/bids/bidOnAProduct`,
        config('POST', {productId, maxBid}), onSuccess, onErrors);

export const updateMaxBid = (productId, maxBid, onSuccess, onErrors) =>
    appFetch(`/bid/bids/updateMaxBid`,
        config('POST', {productId, maxBid}), onSuccess, onErrors);

export const insertProduct = (categoryId, name, description,
                              bidMinutesDuration, initialPrice, shippingInfo,
                              onSuccess, onErrors) =>
    appFetch(`/bid/products/insertProduct`,
        config('POST', {categoryId, name, description,
            bidMinutesDuration, initialPrice, shippingInfo}), onSuccess, onErrors);

export const findDoneBids = ({page}, onSuccess) =>
    appFetch(`/bid/bids?page=${page}`, config('GET'), onSuccess);


export const findInsertedProducts = ({page}, onSuccess) =>
    appFetch(`/bid/products?page=${page}`,
        config('GET'), onSuccess);

File: pa-04/backend/src/main/java/es/udc/paproject/backend/model/services/BidService.java
package es.udc.paproject.backend.model.services;

import es.udc.paproject.backend.model.entities.Bid;
import es.udc.paproject.backend.model.entities.Product;
import es.udc.paproject.backend.model.exceptions.*;

import java.math.BigDecimal;

public interface BidService {

    /**ANUNCIAR PRODUCTO*/
    Product insertProduct(Long userId, Long categoryId, String name, String description, int bidMinutesDuration, BigDecimal initialPrice, String shippingInfo)
            throws InstanceNotFoundException;

    /**PUJAR POR UN PRODUCTO*/
    Bid bidOnAProduct(Long userId, Long productId, BigDecimal maxBid)
            throws InstanceNotFoundException, BidTimeExpiredException, BidMinimumPriceException;

    /**CONSULTAR PRODUCTOS ANUNCIADOS*/
    Block<Product> findInsertedProducts(Long userId, int page, int size);

    /**CONSULTAR PUJAS REALIZADAS*/
    Block<Bid> findDoneBids(Long userId, int page, int size);

}
